app/services/events/stores/aggregated_clickhouse_store.rb

```diff
@@ -112,11 +112,32 @@ def prorated_events_values(total_duration)
       end
 
       def last_event
-        # TODO(pre-aggregation): Implement
+        with_retry { events(ordered: true).last }
       end
 
       def grouped_last_event
-        # TODO(pre-aggregation): Implement
+        connection_with_retry do |connection|
+          cte_sql = events_sql(
+            select: [
+              arel_enriched_table[:grouped_by],
+              arel_enriched_table[:decimal_value].as("property"),
+              arel_enriched_table[:timestamp]
+            ]
+          )
+
+          sql = <<-SQL
+            WITH events AS (#{cte_sql})
+
+            SELECT
+              DISTINCT ON(grouped_by) grouped_by,
```

Comment:
I wonder if this is the most efficient way to retrieve the last element per group 😄